src/buildercore/lifecycle.py

```diff
@@ -283,3 +281,9 @@ def _ec2_connection(stackname):
 
 def _rds_connection(stackname):
     return boto_conn(stackname, 'rds')
+
+def route53conn():
+    """returns a boto2 route53 connection.
+    route53 for boto3 is *very* poor and much too low-level with no 'resource' construct (yet?). It should be avoided"""
+    import boto # will only ever be imported once
```

Comment:
I agree with the isolation but I wouldn't hide `boto` inside a method, imports should be visible at the top of the file?

Reply:
In the future, this could be ported to https://boto3.readthedocs.io/en/latest/reference/services/route53.html#Route53.Client.change_resource_record_sets

Reply:
as per my comment, I don't think the boto3 route53 interface is up to scratch and `change_resource_record_sets` seems far too low level to be switching *to*. I spent some time trying to port the code to boto3 but the translation was just ridiculous. You can't select a specific zone for example, only list all zones with a preference as to which one should be first, if it exists at all. So instead of `get_zone(zone_name)` you need to fetch a list of summaries, ensure the first record has a zone value that matches the one given, and then another call to `get_hosted_zone` using the `Id` from the summary result you just fetched. And then this zone dict has to be pulled apart to get the resource records, that exhibit similar behaviour (no `get_a` methods here) in order to build up an elaborate record for `change_resource_record_sets`.

I reverted most of the code I had and stuck with the boto2 version. It's stable, tested and the interface is convenient.

Reply:
:man_shrugging: sometimes you eat the SDK and sometimes the SDK eats you. I think later on I'll try to move this code into a `buildercore.route53` module which could become the only module including boto2.

Reply:
(ha)

I've been thinking about this actually and it was the reason I created a `sns.py` module. `core.py` and `bootstrap.py` could have big chunks of logic moved out into discrete AWS modules like `sns` or `ec2` or `cloudformation`. The more generic functions I've probably labelled with a TODO.

As core and bootstrap get larger because we're accommodating more stack features it might be worthwhile to do an experimental refactor and see how well it copes.